docs/how-to-guides/creating-dao.md

```diff
@@ -0,0 +1,333 @@
+	TotalPowerAtHeight(height int64) uint64
+}
+```
+
+There is only one implementation currently, `gno.land/p/demo/teritori/dao_voting_group`, it's a wrapper around a gno group, providing a membership-based voting power definition
+
+### Proposal Modules
+
+A proposal module (`gno.land/p/demo/teritori/dao_interfaces.IProposalModule`) is responsible for:
+- Receiving proposals, the proposal type is defined by the module
+- Managing the proposals lifecycle
+- Tallying votes, the vote type is defined by the module and the associated voting power is queried from the voting module
+- Executing proposals once they are passed
+
+**Interface Definition:**
+```go
+type IProposalModule interface {
+	Core() IDAOCore
+	Info() ModuleInfo
+	ConfigJSON() string
+	Render(path string) string
+	Execute(proposalID int)
+	VoteJSON(proposalID int, voteJSON string)
+	ProposeJSON(proposalJSON string) int
+	ProposalsJSON(limit int, startAfter string, reverse bool) string
+	ProposalJSON(proposalID int) string
+}
+```
+
+There is only one implementation currently, `gno.land/p/demo/teritori/dao_proposal_single`, providing a yes/no/abstain vote model with quorum and threshold
+
+### Message handlers
+
+Proposals actions are encoded as objects implementing `gno.land/p/demo/teritori/dao_interfaces.ExecutableMessage`
+```go
+type ExecutableMessage interface {
+	ujson.JSONAble
+	ujson.FromJSONAble
+
+	String() string
+	Type() string
+}
+```
+
+They are deserialized and executed by message handlers implementing `gno.land/p/demo/teritori/dao_interfaces.MessageHandler`
+```go
+type MessageHandler interface {
+	Execute(message ExecutableMessage)
+	MessageFromJSON(ast *json.Node) ExecutableMessage
+	Type() string
+}
+```
+
+Message handlers are registered at core creation and new message handlers can be registered via proposals to extend the DAO capabilities
+
+## Practical Implementation
+
+### Setting Up Your Workspace
+
+Sooo, let's create a new realm
+
+```
+git clone https://github.com/TERITORI/gno.git gno-dao-tutorial
+cd gno-dao-tutorial
+git checkout teritori-unified
+mkdir examples/gno.land/r/demo/my_dao
+```
+
+### Creating the Voting Module
+
+We will start by instantiating a voting module
+
+1. **Initialize the Factory**
+
+Modules instantiation uses the factory pattern in case the module needs to access the core
+
+`examples/gno.land/r/demo/my_dao/my_dao.gno`
+```go
+package my_dao
+
+import (
+    "gno.land/p/demo/teritori/dao_interfaces"
+)
+
+func init() {
+    votingModuleFactory := func(core dao_interfaces.IDAOCore) {
+   
+    }
+}
+```
+
+2. **Instantiate the module**
+
+`examples/gno.land/r/demo/my_dao/my_dao.gno`
+```go
+package my_dao
+
+import (
+    "gno.land/p/demo/teritori/dao_interfaces"
+    "gno.land/p/demo/teritori/dao_voting_group" // <- new
+)
+
+func init() {
+    var group *dao_voting_group.VotingGroup // <- new
+
+    votingModuleFactory := func(core dao_interfaces.IDAOCore) {
+        group = dao_voting_group.NewVotingGroup() // <- new
+    }
+}
```

Comment:
It's not crucial, but what do you think about marking changes in the code snippet like diff? 

For example, line that are added or modifed would be marked with a `+` sign, and lines that are deleted would be marked with a `-` sign. The downside in this case would be that we cannot use syntax highlighting.

Example:

```diff
import (
    "gno.land/p/demo/teritori/dao_interfaces"
+    "gno.land/p/demo/teritori/dao_voting_group"
)

func init() {
+    var group *dao_voting_group.VotingGroup

    votingModuleFactory := func(core dao_interfaces.IDAOCore) {
+        group = dao_voting_group.NewVotingGroup()
    }
}
```

Reply:
I prefer to keep syntax highlighting and pastability